src/net.cpp

```diff
@@ -2634,9 +2634,10 @@ void DumpBanlist()
 
     CBanDB bandb;
     banmap_t banmap;
+    CNode::SetBannedSetDirty(false);
```

Comment:
What about doing the whole `bandb.Write(ban map)` & `CNode::SetBannedSetDirty(false);` while holding `LOCK(cs_setBanned);`?
Otherwise there will always be a possible race condition that someone is adding a ban after  `CNode::SetBannedSetDirty(false);` and before `bandb.Write(banmap)` or `CNode::GetBanned(banmap)`.